doc/source/text.rst

```diff
@@ -201,9 +207,92 @@ and optional groups like
 
 .. ipython:: python
 
-   pd.Series(['a1', 'b2', '3']).str.extract('(?P<letter>[ab])?(?P<digit>\d)')
+   pd.Series(['a1', 'b2', '3']).str.extract('([ab])?(\d)')
+
+can also be used. Note that any capture group names in the regular
+expression will be used for column names; otherwise capture group
+numbers will be used.
+
+Extracting a regular expression with one group returns a DataFrame
+with one column if ``expand=True``, and a Series if ``expand=False``.
+
+.. ipython:: python
+
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=True)
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=False)
+
+Calling ``index.str.extract`` on a regex with exactly one capture
+group returns an ``Index`` if ``expand=False`` and a ``DataFrame``
+with one column if ``expand=True``.
+
+.. ipython:: python
+
+   s = pd.Series(["a1", "b2", "c3"], ["A11", "B22", "C33"])
+   s
+   s.index.str.extract("(?P<letter>[a-zA-Z])", expand=False)
+   s.index.str.extract("(?P<letter>[a-zA-Z])", expand=True)
+
+Calling ``index.str.extract`` on a regex with more than one capture
+group returns a ``DataFrame`` if ``expand=True`` and raises
+``ValueError`` if ``expand=False``.
+
+.. ipython:: python
+
+   s.index.str.extract("(?P<letter>[a-zA-Z])([0-9]+)", expand=True)
+   s.index.str.extract("(?P<letter>[a-zA-Z])([0-9]+)", expand=False)
+
+The table below summarizes the behavior of ``extract(expand=False)``
+(input subject in first column, number of groups in regex in
+first row)
+
++--------+---------+------------+
+|        | 1 group | >1 group   |
++--------+---------+------------+
+| Index  | Index   | ValueError |
```

Comment:
we are defering ``expand=True`` on an ``Index`` -> ``MultiIndex`` for later, yes?